spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/autoconfig/TraceAutoConfiguration.java

```diff
@@ -83,7 +88,9 @@ Tracing tracing(
 			Reporter<zipkin2.Span> reporter, Sampler sampler, ErrorParser errorParser,
 			SleuthProperties sleuthProperties) {
 		Tracing.Builder builder = Tracing.newBuilder().sampler(sampler)
-				.errorParser(errorParser).localServiceName(serviceName)
+				.errorParser(errorParser)
```

Comment:
the default `default` service name is coming from Spring. I don't think that we should hardcode it as you did that

Reply:
It's hardcoded in the @Value expression in this very file already

```
@Value("${spring.zipkin.service.name:${spring.application.name:default}}") String serviceName
```

Reply:
The default value that comes from Spring Boot is "".  So, this will likely never get the value of "default" with that @Value expression.

Reply:
Oh, indeed it is. I wonder why it's resolved to an empty String. I think we should investigate that first

Reply:
Spring boot sets it to empty string.  It's never null.

Reply:
Yeah, I see that. So how about instead of:

`"".equals(serviceName) ? DEFAULT_SERVICE_NAME : serviceName`

we would have

`StringUtils.isEmpty(serviceName) ? DEFAULT_SERVICE_NAME : serviceName`

WDYT?

Reply:
That'll work.  I didn't realize that we had that at our disposal, but that's certainly more readable.  You want me to fix this PR?

Reply:
Yeah, if you could. `StringUtils` from `org.springframework.utils` AFAIR

Reply:
How about this (I'm an Apache Commons committer, so I couldn't resist ;)?
```
StringUtils.defaultIfEmpty(serviceName, DEFAULT_SERVICE_NAME)
```

Reply:
Let's use the Spring one please.